lib/pages/chat/chat.dart

```diff
@@ -1304,6 +1317,114 @@ class ChatController extends State<Chat>
     );
   }
 
+  void onHover(bool isHovered, int index, Event event) {
+    if (timeline!.events[index - 1].eventId == event.eventId &&
+        responsive.isDesktop(context) &&
+        !selectMode &&
+        !openingPopupMenu.value) {
+      focusHover.value = isHovered ? event.eventId : null;
+    }
+  }
+
+  void _handleStateContextMenu() {
+    openingPopupMenu.value = !openingPopupMenu.value;
+  }
+
+  List<ContextMenuItemChatAction> listHorizontalActionMenuBuilder() {
+    final listAction = [
+      ChatHorizontalActionMenu.reply,
+      ChatHorizontalActionMenu.forward,
+      ChatHorizontalActionMenu.more,
+    ];
+    return listAction
+        .map(
+          (action) => ContextMenuItemChatAction(
+            action,
+            action.getContextMenuItemState(),
+          ),
+        )
+        .toList();
+  }
+
+  void handleHorizontalActionMenu(
+    BuildContext context,
+    ChatHorizontalActionMenu actions,
+    Event event,
+  ) {
+    switch (actions) {
+      case ChatHorizontalActionMenu.reply:
+        replyAction(replyTo: event);
+        break;
+      case ChatHorizontalActionMenu.forward:
+        onSelectMessage(event);
+        forwardEventsAction();
+        break;
+      case ChatHorizontalActionMenu.more:
+        handleContextMenuAction(context, event);
+        break;
+    }
+  }
+
+  List<PopupMenuItem> _popupMenuActionTile(
+    BuildContext context,
+    Event event,
+  ) {
+    final listAction = [
+      ChatContextMenuActions.select,
+      ChatContextMenuActions.copyMessage,
+      ChatContextMenuActions.pinMessage,
+      ChatContextMenuActions.forward,
+    ];
+    return listAction.map((action) {
+      return PopupMenuItem(
+        padding: EdgeInsets.zero,
+        child: popupItem(
+          context,
+          action.getTitle(context),
+          iconAction: action.getIcon(),
+          onCallbackAction: () => _handleClickOnContextMenuItem(
+            action,
+            event,
+          ),
+        ),
+      );
+    }).toList();
+  }
+
+  void _handleClickOnContextMenuItem(
+    ChatContextMenuActions action,
+    Event event,
+  ) {
+    switch (action) {
+      case ChatContextMenuActions.select:
+        onSelectMessage(event);
+      case ChatContextMenuActions.copyMessage:
+        onSelectMessage(event);
+        copyEventsAction();
+      case ChatContextMenuActions.pinMessage:
+        onSelectMessage(event);
+        pinEvent();
+      case ChatContextMenuActions.forward:
+        onSelectMessage(event);
+        forwardEventsAction();
+    }
+  }
+
+  void handleContextMenuAction(
+    BuildContext context,
+    Event event,
+  ) {
+    _handleStateContextMenu();
+    openPopupMenuAction(
+      context,
```

Comment:
a lot of context, make me so confused @@

Reply:
ONLY ONE CONTEXT WHEN USER CLICK ON BUTTON